Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,540 @@
+    (a : α) : (∑ i in s, f i) a = ∑ i in s, f i a := by simp
+
+section AddGroup
+
+variable [AddGroup β] [TopologicalAddGroup β] (f g : C_c(α, β))
+
+instance : Neg C_c(α, β) where
+  neg f := { toFun := -f.1
+             continuous_toFun := map_continuous (-f.1)
+             hasCompactSupport' := by simpa [HasCompactSupport, tsupport] using f.2 }
+
+@[simp]
+theorem coe_neg : ⇑(-f) = -f :=
+  rfl
+
+theorem neg_apply : (-f) x = -f x :=
+  rfl
+
+instance : Sub C_c(α, β) where
+  sub f g := { toFun := f.1 - g.1
+               continuous_toFun := map_continuous (f.1 - g.1)
+               hasCompactSupport' := by
+                 simpa [sub_eq_add_neg] using HasCompactSupport.add f.2 (-g).2 }
+
+@[simp]
+theorem coe_sub : ⇑(f - g) = f - g :=
+  rfl
+
+theorem sub_apply : (f - g) x = f x - g x :=
+  rfl
+
+instance : AddGroup C_c(α, β) :=
+  DFunLike.coe_injective.addGroup _ coe_zero coe_add coe_neg coe_sub (fun _ _ => rfl) fun _ _ => rfl
+
+end AddGroup
+
+instance [AddCommGroup β] [TopologicalAddGroup β] : AddCommGroup C_c(α, β) :=
+  DFunLike.coe_injective.addCommGroup _ coe_zero coe_add coe_neg coe_sub (fun _ _ => rfl) fun _ _ =>
+    rfl
+
+instance [Zero β] {R : Type*} [Zero R] [SMulWithZero R β] [SMulWithZero Rᵐᵒᵖ β]
+    [ContinuousConstSMul R β] [IsCentralScalar R β] : IsCentralScalar R C_c(α, β) :=
+  ⟨fun _ _ => ext fun _ => op_smul_eq_smul _ _⟩
+
+instance [Zero β] {R : Type*} [Zero R] [SMulWithZero R β]
+    [ContinuousConstSMul R β] : SMulWithZero R C_c(α, β) :=
+  Function.Injective.smulWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance [Zero β] {R : Type*} [MonoidWithZero R] [MulActionWithZero R β]
+    [ContinuousConstSMul R β] : MulActionWithZero R C_c(α, β) :=
+  Function.Injective.mulActionWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance [AddCommMonoid β] [ContinuousAdd β] {R : Type*} [Semiring R] [Module R β]
+    [ContinuousConstSMul R β] : Module R C_c(α, β) :=
+  Function.Injective.module R ⟨⟨_, coe_zero⟩, coe_add⟩ DFunLike.coe_injective coe_smul
+
+instance [NonUnitalNonAssocSemiring β] [TopologicalSemiring β] :
+    NonUnitalNonAssocSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalSemiring β] [TopologicalSemiring β] :
+    NonUnitalSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalCommSemiring β] [TopologicalSemiring β] :
+    NonUnitalCommSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalNonAssocRing β] [TopologicalRing β] :
+    NonUnitalNonAssocRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
```

Comment:
Why are this and the following instance named, but none of the others are? Let's be consistent. The new automated instance naming scheme is pretty good, so maybe let's not name them manually.